src/rustdoc-internals/htmldocck.md

```diff
@@ -0,0 +1,101 @@
+# `rustdoc` tests
+
+This page is specifically about the `rustdoc` test suite, for other test suites used for testing rustdoc, see [Rustdoc § Tests](../rustdoc.md#tests).
+
+`htmldocck.py` is a custom checker script that uses [XPath] to verify the HTML output of rustdoc.
+
+[XPath]: https://en.wikipedia.org/wiki/XPath
+
+## Directives
+Directives to htmldocck are similar to those given to `compiletest` in that they take the form of `//@` comments.
+
+All `PATH`s in directives are relative to the the rustdoc output directory (`build/TARGET/test/rustdoc/TESTNAME`),
+so it is conventional to use a `#![crate_name = "foo"]` attribute to avoid writing paths.
```

Comment:
That's not really the reason? In my understanding people use that if the crate name is overly long since "TESTNAME" is just the crate name as derived from the file name (if no `#![crate_name]` is present).

Reply:
That's basically what I meant to say, but I seem to have jumbled my words a bit.